src/ucp/core/ucp_ep.h

```diff
@@ -489,6 +489,7 @@ typedef struct ucp_ep_ext {
         ucs_queue_head_t          mid_rdesc_q;    /* Queue of middle fragments, which
                                                      arrived before the first one */
     } am;
+    uct_ep_h                      uct_eps[UCP_MAX_SLOW_PATH_LANES];
```

Comment:
- space line before
- CM should be slow-path lane

Reply:
1) regarding distribute lanes between the arrays:
    I changed the order of lanes insertion, such that all fast lanes will be inserted first.
    if there are more than 5 fast lanes, we'll fail.
    Slow path lanes will be inserted on the first array as long as there's free space there, o/w in the second array.

2) CM lane is now treated as slow path.
3) Making this value configurable is a little complicated, because it has some external constraints, such as lane bit map (which is only 8 bit today).

Reply:
> CM lane is now treated as slow path.

As I see it's still at position `0` which is "fast path"

> I changed the order of lanes insertion, such that all fast lanes will be inserted first.

@shasson5 pls point me to the new logic of lanes distribution which I missed

> Making this value configurable is a little complicated, because it has some external constraints, such as lane bit map (which is only 8 bit today).

and why lane bit map cannot be increased let's say up to 64, same as md map?

Reply:
> and why lane bit map cannot be increased let's say up to 64, same as md map?
not sure what will be the consequences. @yosefe WDYT?

> As I see it's still at position `0` which is "fast path"
right, because at first I thought it should be fast path, will be changed in next commit.

> pls point me to the new logic of lanes distribution which I missed
In the next commit, I will change `ucp_wireup_search_lanes`, so that the first protocols will be fast path

Reply:
> currently it's on a local branch, will be pushed shortly

this is the most interesting part :) changes where could change this PR

Reply:
IMO better increase to more than 8 lanes (and increase the bitmap) in next pr